src/python/pants/backend/helm/target_types.py

```diff
@@ -482,12 +484,56 @@ class HelmDeploymentValuesField(DictStringToStringField):
         """
     )
 
+    @memoized_method
+    @deprecated("2.19.9.dev0", start_version="2.18.0rc0")
+    def format_with(
+        self, interpolation_context: InterpolationContext, *, ignore_missing: bool = False
+    ) -> dict[str, str]:
+        source = InterpolationContext.TextSource(
+            self.address,
+            target_alias=HelmDeploymentTarget.alias,
+            field_alias=HelmDeploymentValuesField.alias,
+        )
+
+        def format_value(text: str) -> str | None:
+            try:
+                return interpolation_context.format(
+                    text,
+                    source=source,
+                )
+            except InterpolationError as err:
+                if ignore_missing:
+                    return None
+                raise err
+
+        result = {}
+        default_curr_value: dict[str, str] = {}
+        current_value: Mapping[str, str] = self.value or default_curr_value
+        for key, value in current_value.items():
+            formatted_value = format_value(value)
+            if formatted_value is not None:
+                result[key] = formatted_value
+
+        if result != current_value:
+            warn_or_error(
+                "2.19.0.dev0",
+                "Using the {env.VAR_NAME} interpolation syntax",
+                "Use the new `f\"prefix-{env('VAR_NAME')}\" syntax for interpolating values from environment variables.",
+                start_version="2.18.0rc0",
```

Comment:
```suggestion
                start_version="2.18.0.dev0",
```

Reply:
What does the other deprecations provide that this one doesn't cover?

Seems to me that this one would catch actual use only which seems preferable.